docs/components/connectors/protocol/soap.md

```diff
@@ -0,0 +1,211 @@
+
+You have to enter the **XML template context** value, e.g.
+`={paramValue: 1234567890}`.
+
+You have to enter the **Namespaces** value, e.g.
+`={"camunda":"http://my.service.com/webservicesserver/"}`.
+
+#### XML compatible JSON
+
+When **XML compatible JSON** is chosen, you have to enter the **JSON definition**, e.g.
+
+```xml
+<camunda:Object01>
+  <camunda:Object02>{{myObjectValue}}</camunda:Object02>
+</camunda:Object01>
+```
+
+You have to enter the **Namespaces** value, e.g.
+`={"camunda":"http://my.service.com/webservicesserver/"}`.
+
+## Output mapping
+
+### Result variable
+
+You can export a complete response from an SOAP call into a dedicated variable accessible anywhere in a process.
+To do so, just input a variable name in the **Result variable** field. We recommend using a unique name to avoid
+variables being overwritten.
+
+## Usage examples
+
+### Example 1
+
+For example, you would like to send a following SOAP request:
+
+URL: `https://myservice:8888/webservice.wso`
+
+Body:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+  <soap:Body>
+    <Object01>
+      <Object02>12345</Object02>
+    </Object01>
+  </soap:Body>
+</soap:Envelope>
+```
+
+In order to do so, in your BPMN diagram, set the field **Service URL** as `https://myservice:8888/webservice.wso`, and **SOAP body** as
+
+```json
+{
+  "Object01": {
+    "Object02": 12345
+  }
+}
+```
+
+### Example 2: pre-defined namespaces
+
+Consider a namespace is defined within your objects, and you wish to send the following request:
+
+URL: `https://myservice:8888/webservice.wso`
+
+Body:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+  <soap:Body>
+    <Object01 mlns="http://www.my.namespace.com/namespace/">
+      <Object02>12345</Object02>
+    </Object01>
+  </soap:Body>
+</soap:Envelope>
+```
+
+In order to do so, in your BPMN diagram, set the field **Service URL** as `https://myservice:8888/webservice.wso`, and **SOAP body** as
+
+```json
+{
+  "ns:Object01": {
+    "ns:Object02": 12345
+  }
+}
+```
+
+Please, pay attention, that here we introduced a new `ns:` prefix. Prefix, can be any arbitrary string, that is not defined as namespace.
+
+Now, you'll need to associate a namespace. You can do it by setting the following value at the **Namespaces** field.
+For the given example, it should be set as:
+
+```json
+{
+  "ns": "http://www.my.namespace.com/namespace/"
+}
+```
+
+### Example 3: using templates
+
+As an alternative, you can use templates to send SOAP messages.
+
+URL: `https://myservice:8888/webservice.wso`
+
+Body:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+  <soap:Body>
+    <Object01>
+      <Object02>12345</Object02>
+    </Object01>
+  </soap:Body>
+</soap:Envelope>
+```
+
+For that, set the **SOAP body** dropdown to **Template**.
```

Comment:
"For that, set the SOAP body dropdown to Template." could be "To do this, set the SOAP body dropdown to Template."